encoder/mjpeg_encoder.cpp

```diff
@@ -39,11 +39,10 @@ MjpegEncoder::~MjpegEncoder()
 		std::cerr << "MjpegEncoder closed" << std::endl;
 }
 
-void MjpegEncoder::EncodeBuffer(int fd, size_t size, void *mem, unsigned int width, unsigned int height,
-								unsigned int stride, int64_t timestamp_us)
+void MjpegEncoder::EncodeBuffer(int fd, size_t size, void *mem, StreamInfo const &info, int64_t timestamp_us)
 {
 	std::lock_guard<std::mutex> lock(encode_mutex_);
-	EncodeItem item = { mem, width, height, stride, timestamp_us, index_++ };
+	EncodeItem item = { mem, info.width, info.height, info.stride, timestamp_us, index_++ };
```

Comment:
Might we want simply to put the whole "info" into the EncodeItem? I guess it doesn't really make any difference, it's not like we're going to do anything with the pixel format or colour space.